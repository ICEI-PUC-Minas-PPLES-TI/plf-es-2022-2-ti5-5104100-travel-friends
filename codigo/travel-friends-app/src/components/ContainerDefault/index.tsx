import React, { ReactNode } from "react";
import { StatusBar, View } from "react-native";

import theme from "../../utils/theme";

import { styles } from "./styles";

type StatusBarProps = {
  children: ReactNode;
  hiden?: boolean;
  isBlue?: boolean;
};

export default function ContainerDefault({
  children,
  hiden,
  isBlue,
}: StatusBarProps) {
  return (
    <View style={[styles.container]}>
      <StatusBar
        barStyle="light-content"
        backgroundColor={isBlue ? theme.colors.primary : theme.colors.shape}
        hidden={hiden}
      />

      {children}
    </View>
  );
}
