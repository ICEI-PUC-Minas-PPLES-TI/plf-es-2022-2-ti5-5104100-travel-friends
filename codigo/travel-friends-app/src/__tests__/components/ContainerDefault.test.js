import React from "react";
import { View } from "react-native";
import renderer from "react-test-renderer";

import ContainerDefault from "../../components/ContainerDefault";

describe("<ContainerDefault />", () => {
  it("has 1 child", () => {
    const tree = renderer
      .create(<ContainerDefault children={<View></View>} />)
      .toJSON();
    expect(tree.children.length).toBe(1);
  });
  it("renders correctly", () => {
    const tree = renderer
      .create(<ContainerDefault children={<View></View>} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
