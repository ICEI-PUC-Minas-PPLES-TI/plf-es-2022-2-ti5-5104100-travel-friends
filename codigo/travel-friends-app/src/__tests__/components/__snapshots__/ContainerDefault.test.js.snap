// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContainerDefault /> renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View />
  </View>
</View>
`;
